### PR TITLE
Transfers the core docs from `aragon/builders-portal` to `aragon/core`

### PR DESCRIPTION
**Pull request by Michael-A-Heuer**
## Description

This PR transfers the core docs from the `aragon/builders-portal` to the `aragon/core` repo

The files
- `.github/workflows/documentation-update.yml`
- `packages/contracts/hardhat.config.ts`
- `packages/contracts/CHANGELOG.md`

were modified to facilitate the transfer. All other files (`.md`) were copied over from the `aragon/builders-portal` commit [0b906b](https://github.com/aragon/builders-portal/tree/0b906b1c17b1a80144504b013bf03d43ac1ff249).

Task: [APP-1069](https://aragonassociation.atlassian.net/browse/APP-1069)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: .github/workflows/documentation-update.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
           path: builders-portal
           token: ${{ secrets.ARABOT_PAT }}
       - name: Copy docs
-        run: cp -R packages/contracts/docs/generated/* $GITHUB_WORKSPACE/builders-portal/docs/smart-contracts/
+        run: cp -R packages/contracts/docs/core/* $GITHUB_WORKSPACE/builders-portal/docs/core/
```
Comment (mathewmeconry) [inline .github/workflows/documentation-update.yml:32]: don't forget to delete the folder `smart-contracts` in the builder-portal repo.
could you please add a `rm -rf` to the folder, so we can remove outdated files too.

**Review by mathewmeconry — COMMENTED** [review]
should this also commit the docs back to this repo?

File: .github/workflows/documentation-update.yml (hunk 1 above)
Comment (Michael-A-Heuer) [reply .github/workflows/documentation-update.yml]: Fixed

**Comment by Michael-A-Heuer** [comment]
> should this also commit the docs back to this repo?

I think this doesn't hurt. On the other hand, the state of `aragon/builders-portal` is currently correct and only needs to be updated if we change something in the core repo from now on.

**Review by mathewmeconry — DISMISSED** [review]

**Comment by Michael-A-Heuer** [comment]
> > > should this also commit the docs back to this repo?
> > 
> > 
> > I think this doesn't hurt. On the other hand, the state of `aragon/builders-portal` is currently correct and only needs to be updated if we change something in the `aragon/core` repo from now on.
> 
> then add another step where arabot commits the changes back to the develop branch in this repo. otherwise the docs commited here won't be up to date and the same as in `aragon/builders-portal` the reason for this is that the workflow generates the documents and then commits it to the builders-portal, where the docs commit in this repo are updated manually

Ah, I think I understand now what you mean. l think the right behavior is to ignore the `packages/contracts/docs/core/03-reference-guide` via `.gitignore` as this is auto-generated from this repo.

**Review by mathewmeconry — APPROVED** [review]

**Pull request merged**